### PR TITLE
common/str_map: fix trim() on empty string

### PR DESCRIPTION
**Pull request by liewegas**
If was pass an empty string (e.g., with something like
get_str_map("M= T= P=") we end up with a (size_t)-1 for end.

http://tracker.ceph.com/issues/38329

### CODE REVIEW

**Comment by neha-ojha** [comment]
This is related:

 47/167 Test  #57: unittest_str_map ........................***Failed    0.65 sec
Running main() from gmock_main.cc
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from str_map
[ RUN      ] str_map.json
[       OK ] str_map.json (0 ms)
[ RUN      ] str_map.plaintext
/home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/common/test_str_map.cc:63: Failure
      Expected: "val1"
To be equal to: str_map["key1"]
      Which is: ""

**Comment by badone** [comment]
Nit: Could you fix the grammatical error in the commit message?

**Comment by badone** [comment]
Jenkins retest this please

**Comment by liewegas** [comment]
fixed!  I had the if condition reversed. 

hoping this fixes it.. the behavior looks undefined but it doesn't crash on my local machine.

File: src/common/str_map.cc [hunk 1]
```diff
@@ -57,6 +57,9 @@ int get_json_str_map(
   return 0;
 }
 string trim(const string& str) {
+  if (str.size()) {
```
Comment (tchaikov) [inline src/common/str_map.cc]: s/size/empty/

Comment (neha-ojha) [reply src/common/str_map.cc]: this seems to be cause for the make check failure

Comment (badone) [reply src/common/str_map.cc]: Groan. I thought that was what @liewegas meant by the push and "I had the if condition reversed". If you change str.size() to !str.size() (or change size() for empty()) it all works. I guess he meant to change it but somehow didn't.

Comment (badone) [inline src/common/str_map.cc]: Serves me right for making assumptions.

Comment (tchaikov) [reply src/common/str_map.cc]: or we can just put 
```c++
  return boost::algorithm::trim_copy_if(str,
                                        [](unsigned char c) {
                                          return std::isspace(c);
                                        });
```

without involving `std::locale`.

**Comment by liewegas** [comment]
ugh, yeah, i forgot the -a to git command --amend.  repushed

**Comment by tchaikov** [comment]
```
  1/170 Test   #6: run-tox-mgr-insights ....................***Failed    3.91 sec
py27 create: /home/jenkins-build/build/workspace/ceph-pull-requests/build/insights/py27
py27 installdeps: pytest, mock
ERROR: invocation failed (exit code 1), logfile: /home/jenkins-build/build/workspace/ceph-pull-requests/build/insights/py27/log/py27-1.log
ERROR: actionid: py27
msg: getenv
cmdargs: '/home/jenkins-build/build/workspace/ceph-pull-requests/build/insights/py27/bin/python -m pip install pytest mock'

DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting pytest
  Could not find a version that satisfies the requirement pytest (from versions: )
No matching distribution found for pytest

ERROR: could not install deps [pytest, mock]; v = InvocationError('/home/jenkins-build/build/workspace/ceph-pull-requests/build/insights/py27/bin/python -m pip install pytest mock (see /home/jenkins-build/build/workspace/ceph-pull-requests/build/insights/py27/log/py27-1.log)', 1)
___________________________________ summary ____________________________________
ERROR:   py27: could not install deps [pytest, mock]; v = InvocationError('/home/jenkins-build/build/workspace/ceph-pull-requests/build/insights/py27/bin/python -m pip install pytest mock (see /home/jenkins-build/build/workspace/ceph-pull-requests/build/insights/py27/log/py27-1.log)', 1)
```

retest this please.

**Comment by liewegas** [comment]
retest this please

**Pull request merged**